### PR TITLE
[MWAR-453] Require Java 8

### PR DESCRIPTION
**Pull request by slachiewicz**

### CODE REVIEW

**Comment by dennisl** [comment]
@slachiewicz Would you mind rebasing your branch to master, now that MWAR-444 and MWAR-450 has been merged. I don't trust myself to do that right now.

**Comment by bmarwell** [comment]
@dennisl can you drop the commit which merges master into this branch and do a rebase instead please?

File: src/test/java/org/apache/maven/plugins/war/AbstractWarMojoTest.java [hunk 1]
```diff
@@ -74,7 +74,7 @@ protected void configureMojo( AbstractWarMojo mojo, List<String> filters, File c
             new DefaultMavenExecutionRequest().setSystemProperties( System.getProperties() ).setStartTime( new Date() );
 
         MavenSession mavenSession =
-            new MavenSession( (PlexusContainer) null, (RepositorySystemSession) null, request, null );
+            new MavenSession( null, null, request, null );
```
Comment (bmarwell) [inline src/test/java/org/apache/maven/plugins/war/AbstractWarMojoTest.java:77]: Are you sure this is a good idea? Imagine a new constructor came  up with four other parameters.

File: src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java [hunk 2]
```diff
@@ -111,7 +111,7 @@ public void testDefaultOverlay()
             assertOverlayedFile( webAppDirectory, "overlay-one", "login.jsp" );
 
             // Ok now check that there is no more files/directories
-            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] { MANIFEST_PATH } );
+            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] {MANIFEST_PATH} );
```
Comment (bmarwell) [inline src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java:114]: That is not a Java 8 change.

File: src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java [hunk 3]
```diff
@@ -149,7 +149,7 @@ public void testDefaultOverlays()
             assertOverlayedFile( webAppDirectory, "overlay-two", "admin.jsp" );
 
             // Ok now check that there is no more files/directories
-            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] { MANIFEST_PATH } );
+            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] {MANIFEST_PATH} );
```
Comment (bmarwell) [inline src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java:152]: That is not a Java 8 change.

File: src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java [hunk 4]
```diff
@@ -293,7 +293,7 @@ private void assertScenariOne( String testId, File webAppDirectory )
             assertOverlayedFile( webAppDirectory, "overlay-full-1", "WEB-INF/lib/c.jar" );
 
             // Ok now check that there is no more files/directories
-            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] { MANIFEST_PATH } );
+            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] {MANIFEST_PATH} );
```
Comment (bmarwell) [inline src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java:296]: That is not a Java 8 change.

File: src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java [hunk 5]
```diff
@@ -432,7 +432,7 @@ public void testOverlaysIncludesExcludesWithMultipleDefinitions2()
             assertOverlayedFile( webAppDirectory, "overlay-full-2", "WEB-INF/lib/c.jar" );
 
             // Ok now check that there is no more files/directories
-            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] { MANIFEST_PATH } );
+            final FileFilter filter = new FileFilterImpl( webAppDirectory, new String[] {MANIFEST_PATH} );
```
Comment (bmarwell) [inline src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java:435]: That is not a Java 8 change.

File: src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java [hunk 6]
```diff
@@ -362,7 +362,7 @@ public void testOverlaysIncludesExcludesWithMultipleDefinitions()
...
```
Comment (bmarwell) [inline src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java:365]: That is not a Java 8 change.

File: src/main/java/org/apache/maven/plugins/war/packaging/AbstractWarPackagingTask.java [hunk 7]
```diff
@@ -342,14 +344,16 @@ protected boolean copyFile( WarPackagingContext context, File source, File desti
     {
         context.addResource( targetFilename );
 
-        if ( onlyIfModified && destination.lastModified() >= source.lastModified() )
+        BasicFileAttributes readAttributes = Files.readAttributes( source.toPath(), BasicFileAttributes.class );
+        if ( onlyIfModified && destination.lastModified() >= readAttributes.lastModifiedTime()
+                .toMillis() )
```
Comment (bmarwell) [inline src/main/java/org/apache/maven/plugins/war/packaging/AbstractWarPackagingTask.java:349]: What is the advantage to use another API here?

**Review by bmarwell — CHANGES_REQUESTED** [review]
Some open questions, but changes requested mainly due to merge commit.

File: src/test/java/org/apache/maven/plugins/war/AbstractWarMojoTest.java (hunk 1 above)
Comment (slachiewicz) [reply src/test/java/org/apache/maven/plugins/war/AbstractWarMojoTest.java:77]: Yes i can revert it. intellij suggested this change

File: src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java (hunk 5 above)
Comment (slachiewicz) [reply src/test/java/org/apache/maven/plugins/war/WarOverlaysTest.java:435]: Sure

File: src/main/java/org/apache/maven/plugins/war/packaging/AbstractWarPackagingTask.java (hunk 7 above)
Comment (bmarwell) [reply src/main/java/org/apache/maven/plugins/war/packaging/AbstractWarPackagingTask.java:349]: It is okay for me. I didn't know about it, and it looked a bit harder to read for me at first, but if it is NIO, then big +1.

**Comment by bmarwell** [comment]
@olamy told me the merge commit will go away due to squashing anyway. Forget my previous comment about dropping and rebasing.

File: src/test/java/org/apache/maven/plugins/war/AbstractWarMojoTest.java (hunk 1 above)
Comment (michael-o) [reply src/test/java/org/apache/maven/plugins/war/AbstractWarMojoTest.java:77]: Please revert this unrelated change

**Review by michael-o — CHANGES_REQUESTED** [review]
+ those from Captain Marvel

**Review by michael-o — CHANGES_REQUESTED** [review]
`MANIFEST_PATH` issue is still there.

**Comment by slachiewicz** [comment]
Sorry, all changes was done from scratch and looks like Intellij automatically makes this changes

**Review by bmarwell — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #525

**Pull request merged**